src/easylink/step.py

```diff
@@ -807,12 +819,43 @@ def set_configuration_state(
             The configuration for any implementations to be combined.
         input_data_config
             The input data configuration for the entire pipeline.
+
+        Notes
+        -----
+        A ``TemplatedStep`` is always assigned a :class:`NonLeafConfigurationState`
+        even if it has no multiplicity since (despite having no copies to make) we
+        still need to traverse the sub-``Steps`` to get to the one with a single
+        :class:`~easylink.implementation.Implementation`, i.e. the one with a
+        :class:`LeafConfigurationState`.
         """
-        num_repeats = len(self._get_config(parent_config[self.name]))
-        self.step_graph = self._update_step_graph(num_repeats)
-        self.slot_mappings = self._update_slot_mappings(num_repeats)
-        super().set_configuration_state(
-            parent_config, combined_implementations, input_data_config
+        step_config = parent_config[self.name]
+        expanded_config = self._get_config(step_config)
+        num_repeats = len(expanded_config) if self.config_key in step_config else 1
+        if num_repeats == 1:
```

Comment:
Isn't `num_repeats == 1` if there is only one iteration of the loop (or one parallel) regardless of whether the user provides the `iterate` (or `parallel`) key in the pipeline configuration? I thought this issue was specifically related to the issue where the user doesn't provide `iterate`.

Reply:
Hmm, I did have to modify `num_repeats` to be 1 in the event they didn't pass in the config key (see one line above, `... if self.config_key in step_config else 1`. I now wonder if this breaks if they do include the config_key but only specify a single iteration/parallel.